modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/AppBrandingPreferenceManagementSuccessTest.java

```diff
@@ -0,0 +1,710 @@
+    public void testUpdateRootAppBrandingPreference() throws Exception {
+
+        String body = readResource(UPDATE_ROOT_APP_BRANDING_RESOURCE_FILE)
+                .replace(APP_ID_PLACEHOLDER, rootAppId);
+        Response response = getResponseOfPut(BRANDING_PREFERENCE_API_BASE_PATH, body);
+
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("type", equalTo(APPLICATION_TYPE))
+                .body("name", equalTo(rootAppId))
+                .body("locale", equalTo(DEFAULT_LOCALE));
+
+        assertBrandingPreferences(UPDATE_ROOT_APP_BRANDING_RESOURCE_FILE, response);
+    }
+
+    @Test(dependsOnMethods = {"testUpdateRootAppBrandingPreference"})
+    public void testGetRootAppBrandingAfterRootAppBrandingUpdate() throws Exception {
+
+        Response response = getResponseOfGet(BRANDING_PREFERENCE_API_BASE_PATH + String.format
+                (PREFERENCE_COMPONENT_WITH_QUERY_PARAM, APPLICATION_TYPE, rootAppId, DEFAULT_LOCALE));
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("type", equalTo(APPLICATION_TYPE))
+                .body("name", equalTo(rootAppId))
+                .body("locale", equalTo(DEFAULT_LOCALE));
+
+        assertBrandingPreferences(UPDATE_ROOT_APP_BRANDING_RESOURCE_FILE, response);
+    }
+
+    @Test(dependsOnMethods = {"testGetRootAppBrandingAfterRootAppBrandingUpdate"})
+    public void testResolveRootAppBrandingAfterRootAppBrandingUpdate() throws Exception {
+
+        Response response = getResponseOfGet(BRANDING_PREFERENCE_RESOLVE_PATH +
+                String.format(PREFERENCE_COMPONENT_WITH_QUERY_PARAM, APPLICATION_TYPE, rootAppId,
+                        DEFAULT_LOCALE));
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("type", equalTo(APPLICATION_TYPE))
+                .body("name", equalTo(rootAppId))
+                .body("locale", equalTo(DEFAULT_LOCALE));
+
+        assertBrandingPreferences(UPDATE_ROOT_APP_BRANDING_RESOURCE_FILE, response);
+    }
+
+    @Test(dependsOnMethods = {"testResolveRootAppBrandingAfterRootAppBrandingUpdate"})
+    public void testResolveL1AppBrandingAfterRootAppBrandingUpdate() throws Exception {
+
+        Response response = getResolvedAppBrandingInOrg(l1OrgId, l1AppId);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("type", equalTo(APPLICATION_TYPE))
+                .body("name", equalTo(rootAppId))
+                .body("locale", equalTo(DEFAULT_LOCALE));
+
+        assertBrandingPreferences(UPDATE_ROOT_APP_BRANDING_RESOURCE_FILE, response);
+
```

Comment:
```suggestion
```

Reply:
Resolved with https://github.com/wso2/product-is/pull/20772/commits/1258e75f5c9e9525d851fd5e4143513bfc72946c